### ISSUE TITLE
Callback URL in Google plugin and conflict with Admin Tools

### ISSUE THREAD

**Issue by Sulpher** [akeeba/sociallogin#37]
My site has .htaccess file which is generated by Admin Tools Pro.
I got a problem with Google authentification.
According to this tutorial: https://github.com/akeeba/sociallogin/wiki/Google
We should use the following callback URL to run Google authentification:
http://www.example.com/index.php?option=com_ajax&group=sociallogin&plugin=google&format=raw

I fact I get an error:

Forbidden
You don't have permission to access /index.php on this server.

I found what prevents running the script. These lines in .htaccess generated by Admin Tools:

##### File injection protection -- BEGIN
RewriteCond %{REQUEST_METHOD} GET
RewriteCond %{QUERY_STRING} [a-zA-Z0-9_]=http[s]?:// [OR]
RewriteCond %{QUERY_STRING} [a-zA-Z0-9_]=(\.\.//?)+ [OR]
RewriteCond %{QUERY_STRING} [a-zA-Z0-9_]=/([a-z0-9_.]//?)+ [NC]
RewriteRule .* - [F]
##### File injection protection -- END

If I comment these lines, all works fine and the authentification works correctly. But additional protection is important.

Before choosing Akeeba Social Login I used Slogjn extension which uses the following format:
http://mysite/?option=com_slogin&task=check&plugin=google
It has no conflict with .htaccess generated by Admin Tools.
I suppose the problem occurs due to index.php? in Akeeba Social Login while SLogin does not have index.php?.

Is there a way to fix this conflict with Akeeba Social Login? Your extension just rocks and I'd like to use it.
I suppose that it requires to change the structure of callback URL in the next version of the plugin?

Thank you.

**Comment by nikosdion** [comment]
I just tried it on my live site but I cannot reproduce this issue. I used the [documentation's instructions](https://github.com/akeeba/sociallogin/wiki/Google) for setting up Google login.

Moreover, there is no reason why the file injection protection would be triggered. If you use the documented callback URL there is no file or URL being passed back by Google which would trigger this issue. I suspect that you may have mistyped something?

Also note that using `http://example.com/?option=com_something` and `http://example.com/index.php?option=com_something` is exactly the same since index.php is the default directory index file for Apache (and it's definitely configured as such by Admin Tools when you choose the option to use index.php before index.html).

So, no, the index.php in the callback URL has nothing to do with it. You can, in fact, remove it if you want. 

The problem is that there is a redirection somewhere with a complete URL passed as a URL parameter. Are you sure you are fully updated? I think I have fixed an issue like that in Social Login several months ago.

**Comment by Sulpher** [comment]
Hi @nikosdion ,
I have recorded a video to show this issue.

Please  watch it:
https://youtu.be/QeFCdwu2RVM

**Comment by nikosdion** [comment]
Again, your problem is not what you describe. It is not index.php. It is, as I said:
> The problem is that there is a redirection somewhere with a complete URL passed as a URL parameter. 

The actual problem is that Google calls the callback URL with the URL parameter `scope=https://www.googleapis.com/auth/userinfo.profile+https://www.googleapis.com/auth/userinfo.email` and your .htaccess blocks it.

This is what Google does. I have no control over it. So the only thing you can do is disable the “Protect against common file injection attacks” feature in Admin Tools' .htaccess Maker.

**Comment by Sulpher** [comment]
Maybe is there a way to exclude Google URL of being blocked by mentioned above .htaccess rules?

**Comment by nikosdion** [comment]
In theory? Yes. In practice? It gets really complicated really fast. I'd have to add a way for you to define exempt URLs or, actually, URL RegEx. Then I'd have to write documentation for an esoteric feature and provide support for it. I'm not particularly interested in implementing such a feature.

If you were able to use this kind of Admin Tools feature you would already know what you are doing, .e. you could disable that feature from the .htaccess Maker and add its modified version in the custom code to put at the beginning of the file. But if that was the case we wouldn't be having this discussion, right?

**Comment by Sulpher** [comment]
Both Admin Tools and Akeeba Social Login are products made by you it is logical that both extensions can be used on the same site.
Akeeba Social Login is a free product and you can ignore any support requests. However, if you are interested to get feedback from the community on this product, you can provide support to issues which you think are worthy of it. (and you are doing it)
If I could solve this issue, I would do it without posting this topic. Thereby, I pointed out a problem which I met and attracted your attention as the product developer since it might be useful to other people. Now, If I find a solution, I will share it and do not ignore this issue since the solution can help other people. And by reporting any new issue I found in a hypothetical product I understand I help to make it better. This is how the community works.
